list.go

```diff
@@ -55,15 +55,23 @@ func (list *List) UnmarshalJSON(rawData []byte) error {
 MarshalJSON returns a top level object for the "data" attribute if a single object. In
 all other cases returns a JSON encoded list for "data".
 */
-func (list List) MarshalJSON() ([]byte, error) {
+func (list *List) MarshalJSON() ([]byte, error) {
```

Comment:
Please add a blurb in the above function comment explaining why we're going through the trouble of treating a list as a pointer in this situation.